packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_place_auction_bid.test.ts

```diff
@@ -0,0 +1,788 @@
+    {
+      await fundEllipticPair(alice.container, aProviders.ellipticPair, 10)
+      const aliceColScript = P2WPKH.fromAddress(RegTest, aliceColAddr, P2WPKH).getScript()
+
+      const txn = await aBuilder.loans.placeAuctionBid({
+        vaultId: bobVaultId,
+        index: '00000000',
+        from: aliceColScript,
+        tokenAmount: { token: 2, amount: new BigNumber(535) }
+      }, aliceColScript)
+
+      // Ensure the created txn is correct
+      const outs = await sendTransaction(alice.container, txn)
+      expect(outs[0].value).toStrictEqual(0)
+      expect(outs[1].value).toBeLessThan(10)
+      expect(outs[1].value).toBeGreaterThan(9.999)
+      expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await aProviders.getAddress())
+
+      // Ensure you don't send all your balance away
+      const prevouts = await aProviders.prevout.all()
+      expect(prevouts.length).toStrictEqual(1)
+      expect(prevouts[0].value.toNumber()).toBeLessThan(10)
+      expect(prevouts[0].value.toNumber()).toBeGreaterThan(9.999)
+
+      await alice.generate(1)
+      await tGroup.waitForSync()
+    }
+
+    // end the auction and alice win the bid
+    await bob.generate(36)
+
+    const auctionsAfter = await bob.container.call('listauctions')
+    expect(auctionsAfter).toStrictEqual([
+      {
+        vaultId: bobVaultId,
+        state: 'inLiquidation',
+        loanSchemeId: 'scheme',
+        ownerAddress: bobVaultAddr,
+        liquidationPenalty: 5,
+        liquidationHeight: expect.any(Number),
+        batchCount: 1,
+        batches: [
+          {
+            index: 0,
+            collaterals: ['5000.00000000@DFI', '0.50000000@BTC'],
+            loan: '500.00428078@TSLA'
+          }
+        ]
+      }
+    ])
+
+    const bobColAccEndBid = await bob.rpc.account.getAccount(bobColAddr)
+    // compare to bobColAccAfter ['8900.00000000@DFI', '19.45454546@TSLA']
+    // bob claims back his funds
+    expect(bobColAccEndBid).toStrictEqual(['8900.00000000@DFI', '545.45454546@TSLA'])
+
+    const aliceColAccEndBid = await alice.rpc.account.getAccount(aliceColAddr)
+    expect(aliceColAccEndBid).toStrictEqual(['35000.00000000@DFI', '29999.50000000@BTC', '9465.00000000@TSLA'])
+
+    // ensure interest is freeze in auction
+    await alice.generate(10)
+    const auctionsAfter1 = await bob.container.call('listauctions')
+    expect(auctionsAfter).toStrictEqual(auctionsAfter1)
+  })
+
+  it('should placeAuctionBid on all batches', async () => {
+    // test bob bids on first index
+    {
+      const bobColAccBefore = await bob.rpc.account.getAccount(bobColAddr)
+      expect(bobColAccBefore).toStrictEqual(['8900.00000000@DFI', '545.45454546@TSLA'])
+
+      await fundEllipticPair(bob.container, bProviders.ellipticPair, 10)
+      const bobColScript = P2WPKH.fromAddress(RegTest, bobColAddr, P2WPKH).getScript()
+
+      const txn = await bBuilder.loans.placeAuctionBid({
+        vaultId: bobVaultId,
+        index: '00000000',
```

Comment:
I think we can improve this, use simple number, can auto do number to 4 byte BE hex string within `CPlaceAuctionBid`. That can make it less error prone for downstream SDK consumer.

Can strengthen the test with larger batches size to check the encoding behavior, eg index 33, it has do `00000021`.